### DISCUSSION TITLE
Creating a Tiff with YCbCr colorspace and 4:2:0 sampling-factor

### DISCUSSION THREAD

**Discussion by Senjuro998** [dlemstra/Magick.NET#1778] [Help] [answered]
I'm combining several tiffs into a single multi-page file. Those with CCITT Group 4 Fax encoding go in as is, otherwise they're converted to JPEG compression, YCbCr colorspace, and 4:2:0 sampling-factor. Everything works except for the sampling-factor which always remains 4:4:4 no matter what SetDefine combination is used. Converting an individual file doesn't work either.

Ultimately the goal is to lower the output's file size without resorting to resizing or significantly lowering the image quality so any way to achieve that will be appreciated.

```
using (MagickImageCollection collection = new MagickImageCollection())
{
	foreach (string inputFile in Directory.GetFiles(source, "*.tif"))
	{
		MagickImage image = new MagickImage(inputFile);
		image.Format = MagickFormat.Tif;

		if (image.Compression != CompressionMethod.Group4)
		{
			image.ColorSpace = ColorSpace.YCbCr;
			image.SetCompression(CompressionMethod.JPEG);
			image.Quality = 70;
			//image.Settings.SetDefine("jpeg:sampling-factor", "2x2,1x1,1x1"); doesn't work
			//image.Settings.SetDefine(MagickFormat.Jpeg, "sampling-factor", "4:2:0"); doesn't work
		}

		collection.Add(image);
	}

	TiffWriteDefines tiffSettings = new TiffWriteDefines();
	tiffSettings.PreserveCompression = true;
	tiffSettings.JpegTablesMode = TiffJpegTablesMode.None;
	collection.Write(output, tiffSettings);
}
```

**Comment by dlemstra** [comment, marked as answer]
It looks like ImageMagick always uses the same value for  `TIFFTAG_YCBCRSUBSAMPLING`. It always uses 1x1 for jpeg compression.
